### DISCUSSION TITLE
Can I implement `ResponseError` trait for crate `rustis`?

### DISCUSSION THREAD

**Discussion by bioinformatist** [actix/actix-web#3233] [Ideas] [answered]
As a user-friendly Redis client, `rustis` comes with its own [`Error` enum](https://docs.rs/rustis/latest/rustis/enum.Error.html). I'm considering implementing our `ResponseError` trait for `rustis::Error`. This could attract new users to `actix-web` and provide convenience for projects of our company (currently, I use snafu to customize error types and implement `ResponseError` for them).

**Comment by robjtede** [comment]
Creating newtype/enum wrappers that semantically reflect the meaning of error responses is idiomatic for Actix Web apps.

Would you like to expand on what it is you're proposing?

**Reply by bioinformatist** [reply]
Hi Rob, we're developing web apps with `actix-web` and `rustis` (Redis client) and I've observed that `rustis` has its own [`Error` enum](https://docs.rs/rustis/latest/rustis/enum.Error.html). In order to use `Result<impl Responder, E>` as the return value of handlers, the `actix_web::ResponseError` trait needs to be implemented for `rustis::Error`. I've noticed that the trait has already been implemented for some external types, such as `serde` and `openssl`. Given that Redis will be utilized by numerous applications, I believe this implementation will greatly benefit many developers. May I submit a PR for this?

**Reply by robjtede** [reply, marked as answer]
Ahh I see. No, I would not accept a PR for this. It would be more appropriate to submit this PR to `rustis`. However, I stand by the idea that semantic newtype wrappers are the best method currently available to applications to suit their particular needs.

> the trait has already been implemented for some external types

These external crates are core to the operation of the framework.

**Reply by bioinformatist** [reply]
Got it. Thanks to such a quick reply!
